devito/ir/equations/algorithms.py

```diff
@@ -90,9 +91,11 @@ def generate_implicit_exprs(expressions):
                 sub_dims.append(e.subdomain.implicit_dimension)
                 dims = [d for d in dims if d not in frozenset(sub_dims)]
                 dims.append(e.subdomain.implicit_dimension)
-                grid = list(retrieve_functions(e, mode='unique'))[0].grid
-                processed.extend([i.func(*i.args, implicit_dims=dims) for i in
-                                  e.subdomain._create_implicit_exprs(grid)])
+                if e.subdomain not in found:
```

Comment:
Did we have an MFE relating to this change that could be added to `test_subdomains.py`?

Reply:
honestly, I don't even remember how I bumped into it... what was the issue again? :grimacing:

Reply:
I mean we, for example, process the same subdomain twice... and then what happens?

Reply:
but IIRC I think that some tests now, with all the other changes shipped by this PR,  do fail w/o this patch

Reply:
We were getting unnecessary replication of implicit equations. I think you posted an MFE in development a while back, but would need to dig it out.

Reply:
now addressed via https://github.com/devitocodes/devito/pull/1469/files#diff-7b79cb6a68638992ebb71a66e8faf75f4508c905d6b68a8827830383a693eedfR258